libraries/_resoruce_helpers.rb

```diff
@@ -107,6 +107,42 @@ def cp_r(src, dst)
       ::FileUtils.chown_R(node['knotx']['user'], node['knotx']['group'], dst)
     end
 
+    def get_top_level_distro_directory(tmp_dir_path)
+      Dir.entries(tmp_dir_path).select {
+        |entry| File.directory? ::File.join(tmp_dir_path, entry) and !(entry == '.' || entry == '..')
+      }
+    end
+
+    # Validates Knot.x stack distribution structure
+    # Expected one is:
+    # .
+    # └── ** (any folder name, e.g. `my-stack` or `knotx`)
+    #     ├── conf
+    #     └── lib
+    #
+    # If validation is successfull then returns top folder name
+    def validate_distro_structure(tmp_dir_path)
+      Chef::Log.debug("Validating distro structure of: #{tmp_dir_path}")
+      top_folder = get_top_level_distro_directory(tmp_dir_path)
+      if top_folder.size == 1
+        tmp_lib_path = ::File.join(tmp_dir_path, top_folder, 'lib')
+        tmp_lib_exists = ::File.exist? tmp_lib_path
+
+        tmp_conf_path = ::File.join(tmp_dir_path, top_folder, 'conf')
+        tmp_conf_exists = ::File.exist? tmp_conf_path
+
+        if !(tmp_lib_exists && tmp_conf_exists)
+          Chef::Application.fatal!(
+            "Expected `lib` and `conf` folders inside #{top_folder} not found!"
```

Comment:
Minor thing, but backticks won't be evaluated in this context

Reply:
I don't want to evaluate them, this is just my habit form markdown. I will remove backticks from here.